### PR TITLE
Adding a label to the toggle component for using the WordPress.com na…

### PR DESCRIPTION
**Pull request by spraveenitpro**
#### Changes proposed in this Pull Request

This PR adds a label to the Toggle component available on /domains/manage/:domain/name-servers/:site , it addresses #53003

#### Testing instructions

* Goto /domains/manage/:domain/name-servers/:site
* We should now be able to see a label for the toggle button like this 
[Direct Link](https://d.pr/i/72OzPq+)

Related to #53003
Request review from @tyxla

### CODE REVIEW

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx [hunk 1]
```diff
@@ -31,6 +31,7 @@ class NameserversToggle extends PureComponent {
 						onChange={ this.handleToggle }
 						checked={ this.props.enabled }
 						value="active"
+						label={ this.props.translate( 'On' ) }
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: It's nice that we're using `translate()` to localize the new string. However, looking from the point of view of a translator, I wouldn't be sure how to translate that string without some context. Can you find a way to provide some context for translators?

Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: It's a bummer we're adding a new visual label here. I'd expect that the "Use WordPress.com Name Servers" part works like a label itself, rather than introducing a new label. Not to mention that "On" next to a toggle is redundant - the toggle element is clear enough to visually indicate whether it's on or not.

**Review by tyxla — COMMENTED** [review]
Thanks for working on this one @spraveenitpro!

It would be nice if we don't reintroduce any design changes - we already have a piece of the UI that can function as a label, but it currently doesn't. 

Let me know if you have any questions.

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx (hunk 1 above)
Comment (spraveenitpro) [reply client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: Absolutely makes sense, do not want to add more visual cues, I will see how to remove the visual label

Comment (spraveenitpro) [reply client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: So I checked previous usages of <ToggleControl> and all of them had label(around 20), in one case I found label missing. Now I have enclosed the entire <ToggleControl> within a `label`, let me know if that works. Reference: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx [hunk 2]
```diff
@@ -25,13 +25,16 @@ class NameserversToggle extends PureComponent {
...
-						name="wp-nameservers"
-						onChange={ this.handleToggle }
-						checked={ this.props.enabled }
-						value="active"
-					/>
+					<label>
+						<ToggleControl
+							id="wp-nameservers"
+							name="wp-nameservers"
+							git
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: Oops, this looks like an accidental artifact that should be removed.

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx [hunk 3]
```diff
@@ -25,13 +25,16 @@ class NameserversToggle extends PureComponent {
...
 
 				<form className="name-servers__toggle">
-					<ToggleControl
-						id="wp-nameservers"
-						name="wp-nameservers"
-						onChange={ this.handleToggle }
-						checked={ this.props.enabled }
-						value="active"
-					/>
+					<label>
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: We have a component for labels, is there a reason why we're not using it here?

Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: Also, does using a plain `<label />` make the toggle more accessible? I'm very interested in hearing your thoughts about this.

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx [hunk 4]
```diff
@@ -25,13 +25,16 @@ class NameserversToggle extends PureComponent {
 				</span>
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: It seems like we already have a text that describes the field, but it's not a label. Is there a reason why we're not using that for the label?

**Review by tyxla — CHANGES_REQUESTED** [review]

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx (hunk 1 above)
Comment (tyxla) [reply client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: Thanks, I've left some more feedback in a separate review.

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx (hunk 3 above)
Comment (spraveenitpro) [reply client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: Great point, are you referring to the `Focusable` component?  I am looking for the components [from here.](https://github.com/Automattic/wp-calypso/tree/04c1ce6307017c1abe8ed55be244f320c893ddec/client/components)

Comment (spraveenitpro) [reply client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: From testing with an actual screenreader, it does not seem so, Currently testing with   `<ScreenReaderText>` component.

Comment (tyxla) [reply client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: I was talking about `FormLabel` [here](https://github.com/Automattic/wp-calypso/blob/04c1ce6307017c1abe8ed55be244f320c893ddec/client/components/forms/form-label/index.tsx).

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx (hunk 4 above)
Comment (spraveenitpro) [reply client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: I have added the `FormLabel` component around the `ToggleControl` let me know if I am on the right track.

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx [hunk 5]
```diff
@@ -25,13 +26,15 @@ class NameserversToggle extends PureComponent {
...
 
 				<form className="name-servers__toggle">
-					<ToggleControl
-						id="wp-nameservers"
-						name="wp-nameservers"
-						onChange={ this.handleToggle }
-						checked={ this.props.enabled }
-						value="active"
-					/>
+					<FormLabel className="name-servers__toggle">
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: Is there a reason for adding the same `className` as we have for the `<form />` above?

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx (hunk 4 above)
Comment (tyxla) [reply client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: It is one of the ways to add a label. But it doesn't seem to do the job in terms of accessibility. When using a screen reader I don't see a difference.

**Review by tyxla — COMMENTED** [review]
Thanks for your continuous work here 🙌 

So, I have 2 general suggestions to help push this forward:

1. I think we should find a way that works for when we're using a screen reader, but doesn't introduce any visual changes. We'd like to improve the situation for screen readers, but not affect the design in any way. What have you tried in that direction?
2. Optional: Ideally, the "Use WordPress.com Name Servers" part should be the label - it already is by design, but it doesn't function as a label from an accessibility point of view, nor is it semantically a label. Is this something you've tried out to achieve?

**Comment by spraveenitpro** [comment]
> * I think we should find a way that works for when we're using a screen reader, but doesn't introduce any visual changes. We'd like to improve the situation for screen readers, but not affect the design in any way. What have you tried in that direction?

 I have now used the [ScreenReaderText](https://wpcalypso.wordpress.com/devdocs/packages/components/src/screen-reader-text/README.md) component  and also tested with the screenreader on mac to ensure  it improves accessibility. The test works as expected and is able to read out  "Use WordPress.com Name Servers" for the toggle.

> * Optional: Ideally, the "Use WordPress.com Name Servers" part should be the label - it already is by design, but it doesn't function as a label from an accessibility point of view, nor is it semantically a label. Is this something you've tried out to achieve?

Now, that I have used "Use WordPress.com Name Servers"  with `ScreenReaderText` within `FormLabel` do you think we have met this requirement.

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx (hunk 5 above)
Comment (spraveenitpro) [reply client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: I have removed it now, used some previous examples for reference.

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx (hunk 4 above)
Comment (spraveenitpro) [reply client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: After the most recent commit, I tested a few times with the mac screen reader and it is working.

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx [hunk 6]
```diff
@@ -25,13 +26,18 @@ class NameserversToggle extends PureComponent {
...
 
 				<form className="name-servers__toggle">
-					<ToggleControl
-						id="wp-nameservers"
-						name="wp-nameservers"
-						onChange={ this.handleToggle }
-						checked={ this.props.enabled }
-						value="active"
-					/>
+					<FormLabel>
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: Seems like wrapping everything in a `<FormLabel />` won't do the job, because `ToggleControl` already includes a `<label />`. This means we'll end up with a `<label />` inside a `<label />`, which is not semantic and is an invalid HTML:



See https://html.spec.whatwg.org/multipage/forms.html#the-label-element:phrasing-content-2-3.

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx [hunk 7]
```diff
@@ -25,13 +26,18 @@ class NameserversToggle extends PureComponent {
...
+						<ToggleControl
+							id="wp-nameservers"
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: Something to note here (and I realize it's pre-existing) is that the `id` attribute never makes it further, because the `ToggleControl` doesn't do anything with it.

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx [hunk 8]
```diff
@@ -25,13 +26,18 @@ class NameserversToggle extends PureComponent {
...
+							name="wp-nameservers"
+							onChange={ this.handleToggle }
+							checked={ this.props.enabled }
+							value="active"
+						/>
+						<ScreenReaderText>
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: The `ScreenReaderText` is a step forward for making this toggle more accessible 👍 

There are essentially 2 flaws I see here with this approach:

* We duplicate the `Use WordPress.com Name Servers` string. It would be better if we only had to include it just once in this component, and that would be sufficient for both the visible label and the screen readers.
* The visible label is still disconnected from the toggle. I'd expect that if I click the `Use WordPress.com Name Servers` label, I'd enable/disable the toggle. Note that this should also be evident when navigating with a screen reader.

**Review by tyxla — CHANGES_REQUESTED** [review]
Thanks for keeping up and working to improve this piece 🙌 

I think we're getting closer, but the approach still needs some polishing. 

I've left some more feedback, let me know what you think.

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx (hunk 8 above)
Comment (spraveenitpro) [reply client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: Both the above points are very important:

* I have now avoided the duplication of `Use WordPress.com Name Servers` string by using a constant, hope that is the correct approach. The same constant is used for both the visible label and the screen reader text
* Visible label is now connected to the toggle, if you click the visible label it will enable/disable toggle

### Important:

* Removing the FormLabel for some reason is causing a situation where the ScreenReaderText is not working any more when testing with the voiceover tool in mac, any suggestions on this would be great. 🙏🏼 @tyxla

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx [hunk 9]
```diff
@@ -1,15 +1,17 @@
...
 import { PureComponent } from 'react';
 import { connect } from 'react-redux';
+import FormLabel from 'calypso/components/forms/form-label';
 import { CHANGE_NAME_SERVERS } from 'calypso/lib/url/support';
 import {
 	composeAnalytics,
 	recordGoogleEvent,
 	recordTracksEvent,
 } from 'calypso/state/analytics/actions';
+const NAMESERVERS_TOGGLE_TITLE = 'Use WordPress.com Name Servers';
```
Comment (rafaelgallani) [inline client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: Nice job moving the label to a `const`, but I think they wouldn't be translated like this. 
You also need to add the `__` module (`@wordpress/i18n` package) call to retrieve the translation of the text. 

[Here](https://developer.wordpress.org/reference/functions/__/) are the docs for it. 

There's also a similar example here:

https://github.com/Automattic/wp-calypso/blob/f8c7fe5df453bde9a7582df5c092821baf3ef6a0/client/components/domains/connect-domain-step/constants.ts#L55-L68

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx [hunk 10]
```diff
@@ -1,15 +1,17 @@
...
 
 class NameserversToggle extends PureComponent {
```
Comment (rafaelgallani) [inline client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: Regarding the [previous comment](https://github.com/Automattic/wp-calypso/pull/57374#discussion_r743378849): 

Since the component is already wrapped in the `localize` HOC, I would move the previous title `const` declaration to the instance level (constructor) and use the `translate` prop. Something like: 

```suggestion
class NameserversToggle extends PureComponent {
    constructor( props ){
        super( props );

        this.NAMESERVERS_TOGGLE_TITLE = props.translate( 'Use WordPress.com Name Servers' );
    }
```

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx (hunk 9 above)
Comment (tyxla) [reply client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: Moving from a localized and translated string to non-translated one is definitely a regression as @rafaelgalani recognized.

Furthermore, I don't think this is an instance that requires a getter function like in the constants file in the previous comment. For me, the problem wasn't that we're duplicating the same text twice; but rather that we need to have a separate `span` with a visual label, and then the toggle separated semantically from it, while the label and the toggle are disconnected from each other.

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx (hunk 10 above)
Comment (tyxla) [reply client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: I don't see the need to do this either. Avoiding calling a `translate()` again brings little to no performance improvement, at the cost of a new variable, so I don't think we need to do it.

However I agree that the component already has access to the `translate` prop and can freely use it.

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx [hunk 11]
```diff
@@ -20,18 +22,21 @@ class NameserversToggle extends PureComponent {
 	render() {
 		return (
 			<Card compact className="name-servers__dns">
-				<span className="name-servers__title">
-					{ this.props.translate( 'Use WordPress.com Name Servers' ) }
+				<span className="name-servers__title" onClick={ this.handleToggle }>
+					{ NAMESERVERS_TOGGLE_TITLE }
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: What if we remove this `span` and add the label text to the toggle component? Then style the label to look like this span looked. So essentially, using a sole `ToggleControl` for the purpose. Have you tried that? I believe it should solve all the problems we are facing.

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx (hunk 9 above)
Comment (rafaelgallani) [reply client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: >Furthermore, I don't think this is an instance that requires a getter function like in the constants file in the previous comment.

@tyxla If the const was declared inside the class (constructor, method, etc), then you're right; but if the constant was kept outside React's context, then I think this would be necessary since the translations wouldn't be retrieved as the HOC props were updated.

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx (hunk 11 above)
Comment (spraveenitpro) [reply client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: You are right. I have just done this along with some styling,  please let me know.

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx (hunk 9 above)
Comment (tyxla) [reply client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: @rafaelgalani see https://github.com/Automattic/wp-calypso/pull/57374#discussion_r743654922 - as part of my review I'm suggesting an alternative solution that keeps the string inside the React component, which is why I've suggested that the getters would be unnecessary overkill.

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx (hunk 11 above)
Comment (tyxla) [reply client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: Thanks, @spraveenitpro, I'm going to take a look at the updated code on Monday.

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx [hunk 12]
```diff
@@ -20,18 +22,18 @@ class NameserversToggle extends PureComponent {
...
-				</span>
-
 				<form className="name-servers__toggle">
 					<ToggleControl
 						id="wp-nameservers"
 						name="wp-nameservers"
 						onChange={ this.handleToggle }
 						checked={ this.props.enabled }
 						value="active"
+						label={ this.props.translate( NAMESERVERS_TOGGLE_TITLE ) }
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: This string will not end up being translated properly. The problem comes from how we manage translations, let me elaborate:

1) We use `translate()` and `__()` and other localization functions around our strings we want to translate (e.g. localize).
2) Automated jobs scan the code for new strings using those specific function calls, and consider them as strings for translation. Those are then added to the queue for translation.
3) Once those strings are translated, and we serve the translations together with this code, and display the translated code.

However, there is one important requirement for 1) - it does performs static analysis on the code, and it can't run the code, so `NAMESERVERS_TOGGLE_TITLE` will not be evaluated, and `this.props.translate( NAMESERVERS_TOGGLE_TITLE )` will end up improperly localized and not translated.

So, in a nutshell, we should always hardcode the strings inside the localization functions. Of course, those strings can have dynamic parts (components, variables, etc.) but there's a different syntax for that.

Let me know if that makes sense.

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx [hunk 13]
```diff
@@ -20,18 +22,18 @@ class NameserversToggle extends PureComponent {
...
+						className="name-servers__title"
+						onClick={ this.handleToggle }
 					/>
+					<ScreenReaderText>{ this.props.translate( NAMESERVERS_TOGGLE_TITLE ) }</ScreenReaderText>
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: Furthermore, we can see that both times when `NAMESERVERS_TOGGLE_TITLE` is used, it's in the same component, and even in the same component.

So why not just declare a `const toggleTitle = translate( 'Use WordPress.com Name Servers' );` at the top of the `render()` function and use the `toggleTittle` variable in both places?

File: client/my-sites/domains/domain-management/name-servers/style.scss [hunk 14]
```diff
@@ -5,12 +5,24 @@
 }
 
 .name-servers__dns {
-	.name-servers__toggle {
-		display: inline-block;
-		float: right;
+	display: flex;
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/style.scss]: Do we really need the card to be `flex`?

File: client/my-sites/domains/domain-management/name-servers/style.scss [hunk 15]
```diff
@@ -5,12 +5,24 @@
...
+	flex-direction: column;
 
-		.components-toggle-control .components-base-control__field .components-form-toggle {
-			margin-right: 0;
+	.name-servers__toggle {
+		display: block;
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/style.scss]: Aren't `<form />` tags with `display: block` by default?

File: client/my-sites/domains/domain-management/name-servers/style.scss [hunk 16]
```diff
@@ -5,12 +5,24 @@
...
+		.components-toggle-control .components-base-control__field {
+			display: flex;
+			justify-content: space-between;
+			.components-form-toggle {
+				order: 1;
+			}
+			.components-toggle-control__label {
+				order: 0;
+				border: 0;
+				font-size: 16px;
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/style.scss]: `px` font sizes are deprecated. Is there a suitable variable to use here?

File: client/my-sites/domains/domain-management/name-servers/style.scss [hunk 17]
```diff
@@ -5,12 +5,24 @@
...
+		display: block;
+		.components-toggle-control .components-base-control__field {
+			display: flex;
+			justify-content: space-between;
+			.components-form-toggle {
+				order: 1;
+			}
+			.components-toggle-control__label {
+				order: 0;
+				border: 0;
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/style.scss]: Why is a `border` necessary?

File: client/my-sites/domains/domain-management/name-servers/style.scss [hunk 18]
```diff
@@ -5,12 +5,24 @@
...
 
 .name-servers__dns {
-	.name-servers__toggle {
-		display: inline-block;
-		float: right;
+	display: flex;
+	flex-direction: column;
 
-		.components-toggle-control .components-base-control__field .components-form-toggle {
-			margin-right: 0;
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/style.scss:13]: Is there a reason why we're removing the right margin? This seems to push the toggle a bit to the left.

File: client/my-sites/domains/domain-management/name-servers/style.scss [hunk 19]
```diff
@@ -5,12 +5,24 @@
...
+			display: flex;
+			justify-content: space-between;
+			.components-form-toggle {
+				order: 1;
+			}
+			.components-toggle-control__label {
+				order: 0;
+				border: 0;
+				font-size: 16px;
+				font-style: inherit;
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/style.scss]: Is this necessary?

File: client/my-sites/domains/domain-management/name-servers/style.scss [hunk 20]
```diff
@@ -5,12 +5,24 @@
...
+				font-weight: 400;
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/style.scss]: Why is this necessary?

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx [hunk 21]
```diff
@@ -1,15 +1,17 @@
-import { Card } from '@automattic/components';
+import { Card, ScreenReaderText } from '@automattic/components';
 import { ToggleControl } from '@wordpress/components';
 import { localize } from 'i18n-calypso';
 import PropTypes from 'prop-types';
 import { PureComponent } from 'react';
 import { connect } from 'react-redux';
+import FormLabel from 'calypso/components/forms/form-label';
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: Seems like the `FormLabel` component is no longer in use.

**Review by tyxla — CHANGES_REQUESTED** [review]
Reusing the label from the `ToggleControl` component is a huge step forward, thanks! 🙌 

There are a few bits to iron out still, though - please take a look and let me know if you have any questions.

**Review by tyxla — COMMENTED** [review]
Oh, and also, let's make sure we don't introduce any visual changes in the meantime. There seem to be some spacing differences when comparing to `trunk`:

Before:


After:

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx (hunk 13 above)
Comment (spraveenitpro) [reply client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: This was a major major ahaaa moment, Thank you. 🔥🔥

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx (hunk 12 above)
Comment (spraveenitpro) [reply client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: Completely makes sense, Thank you for this write up. Frankly I did not know this. Now I know. I am beginning to see what I do not know 🙌🏽 Masterclass!!

File: client/my-sites/domains/domain-management/name-servers/style.scss (hunk 14 above)
Comment (spraveenitpro) [reply client/my-sites/domains/domain-management/name-servers/style.scss]: I have removed this also.

File: client/my-sites/domains/domain-management/name-servers/style.scss (hunk 16 above)
Comment (spraveenitpro) [reply client/my-sites/domains/domain-management/name-servers/style.scss]: Using `$font-body` variable

File: client/my-sites/domains/domain-management/name-servers/style.scss (hunk 20 above)
Comment (spraveenitpro) [reply client/my-sites/domains/domain-management/name-servers/style.scss]: Completely cleaned up.

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx (hunk 21 above)
Comment (spraveenitpro) [reply client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: Removed, I think we are almost there.

File: client/my-sites/domains/domain-management/name-servers/style.scss (hunk 16 above)
Comment (tyxla) [reply client/my-sites/domains/domain-management/name-servers/style.scss]: Good choice!

File: client/my-sites/domains/domain-management/name-servers/style.scss [hunk 22]
```diff
@@ -6,14 +6,15 @@
 
 .name-servers__dns {
 	.name-servers__toggle {
-		display: inline-block;
-		float: right;
-
-		.components-toggle-control .components-base-control__field .components-form-toggle {
-			margin-right: 0;
+		.components-toggle-control .components-base-control__field {
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/style.scss]: Do we need the selector specificity here?

File: client/my-sites/domains/domain-management/name-servers/style.scss [hunk 23]
```diff
@@ -6,14 +6,15 @@
...
+			.components-toggle-control__label {
+				font-size: $font-body;
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/style.scss:13]: Good choice of variable for the font size 👍

File: client/my-sites/domains/domain-management/name-servers/style.scss [hunk 24]
```diff
@@ -6,14 +6,15 @@
...
+			}
+			.components-form-toggle__input {
+				align-self: center;
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/style.scss]: This doesn't seem to do anything?

File: client/my-sites/domains/domain-management/name-servers/style.scss [hunk 25]
```diff
@@ -6,14 +6,15 @@
 
 .name-servers__dns {
 	.name-servers__toggle {
-		display: inline-block;
-		float: right;
-
-		.components-toggle-control .components-base-control__field .components-form-toggle {
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/style.scss:12]: This PR still introduces visual changes. Seems like we still need to tinker with the order of the toggle and its label, as well as with the white space:

Production:


This branch:

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx [hunk 26]
```diff
@@ -18,20 +18,21 @@ class NameserversToggle extends PureComponent {
...
 				<form className="name-servers__toggle">
 					<ToggleControl
 						id="wp-nameservers"
 						name="wp-nameservers"
 						onChange={ this.handleToggle }
 						checked={ this.props.enabled }
 						value="active"
+						label={ toggleTitle }
+						className="name-servers__title"
+						onClick={ this.handleToggle }
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: Is the `onClick` necessary? Isn't the `onChange` above sufficient?

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx [hunk 27]
```diff
@@ -18,20 +18,21 @@ class NameserversToggle extends PureComponent {
...
-				</span>
-
 				<form className="name-servers__toggle">
 					<ToggleControl
 						id="wp-nameservers"
 						name="wp-nameservers"
 						onChange={ this.handleToggle }
 						checked={ this.props.enabled }
 						value="active"
+						label={ toggleTitle }
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: Nice work reusing the built-in label inside `ToggleControl` 👏

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx [hunk 28]
```diff
@@ -18,20 +18,21 @@ class NameserversToggle extends PureComponent {
...
+						className="name-servers__title"
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: Is that `className` necessary for anything? If not, perhaps we can just remove it?

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx [hunk 29]
```diff
@@ -18,20 +18,21 @@ class NameserversToggle extends PureComponent {
...
+						onClick={ this.handleToggle }
 					/>
+					<ScreenReaderText>{ toggleTitle }</ScreenReaderText>
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: Is the `<ScreenReaderText />` necessary now that we're using the built-in label from the `ToggleControl` component? Can we just remove it if that's the case?

**Review by tyxla — CHANGES_REQUESTED** [review]
This looks great, thank you! 

There are several more final details we should iron out before shipping it, but we've achieved the most important points now: reusing the built-in label and making it accessible 👏 

I've left feedback and suggestions, let me know if you need any further clarification or help.

Thanks for all the hard work here, you rock 🤘

File: client/my-sites/domains/domain-management/name-servers/style.scss (hunk 23 above)
Comment (spraveenitpro) [reply client/my-sites/domains/domain-management/name-servers/style.scss:13]: I started looking at the code by doing random searches and finally found it.

File: client/my-sites/domains/domain-management/name-servers/style.scss (hunk 24 above)
Comment (spraveenitpro) [reply client/my-sites/domains/domain-management/name-servers/style.scss]: From what I am seeing this is not there anymore.

File: client/my-sites/domains/domain-management/name-servers/style.scss (hunk 25 above)
Comment (spraveenitpro) [reply client/my-sites/domains/domain-management/name-servers/style.scss:12]: This is what I am seeing on this branch:


[Direct Link](https://d.pr/i/M7nr0d+)

From what I am seeing, it looks pretty similar to the production? 🤞🏽

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx (hunk 29 above)
Comment (spraveenitpro) [reply client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: I actually did not know that the ToggleControl Label would suffice, I have removed it now. I also proceeded to test with the Mac voiceover and it works great! besides no console error. You are teaching me `attention to detail`. Thank you !

File: client/my-sites/domains/domain-management/name-servers/style.scss (hunk 25 above)
Comment (tyxla) [reply client/my-sites/domains/domain-management/name-servers/style.scss:12]: It looks much better after the recent changes! There are a couple of minor things to iron out, I'll leave a separate review.

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx (hunk 29 above)
Comment (tyxla) [reply client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: Nice! And yep, I confirmed it's working great now 🎉

File: client/my-sites/domains/domain-management/name-servers/style.scss [hunk 30]
```diff
@@ -6,14 +6,17 @@
...
-			margin-right: 0;
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/style.scss:13]: We still need the `margin-right: 0` here - otherwise, the toggle gets pushed to the left a bit.

File: client/my-sites/domains/domain-management/name-servers/style.scss [hunk 31]
```diff
@@ -6,14 +6,17 @@
...
+		.components-base-control__field {
+			justify-content: space-between;
+			margin-bottom: 0;
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/style.scss]: I'd suggest removing the `margin-bottom: 0`. It adds a bit more spacing at the bottom than what we used to have, but to me, it looks fine and doesn't look so packed. So it's a good compromise IMHO.

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx [hunk 32]
```diff
@@ -18,19 +18,17 @@ class NameserversToggle extends PureComponent {
...
-				</span>
-
 				<form className="name-servers__toggle">
 					<ToggleControl
 						id="wp-nameservers"
 						name="wp-nameservers"
 						onChange={ this.handleToggle }
 						checked={ this.props.enabled }
 						value="active"
+						label={ toggleTitle }
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: Now that `toggleTitle` is used only once, we don't need a variable; we can directly pass `this.props.translate( 'Use WordPress.com Name Servers' )` to the `label` prop.

**Review by tyxla — CHANGES_REQUESTED** [review]
This is working and looking great now, thank you 🚀 

There are a couple of little CSS improvements we can make, and one little code style improvement in the component. 

Once that's addressed, I think this is ready to go. 

Thanks a bunch for your hard work here, @spraveenitpro 🙌

File: client/my-sites/domains/domain-management/name-servers/style.scss (hunk 31 above)
Comment (spraveenitpro) [reply client/my-sites/domains/domain-management/name-servers/style.scss]: I agree with this, just removed it.

File: client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx (hunk 32 above)
Comment (spraveenitpro) [reply client/my-sites/domains/domain-management/name-servers/wpcom-nameservers-toggle.jsx]: Haa...I think we have arrived... 🙌🏽

File: client/my-sites/domains/domain-management/name-servers/style.scss [hunk 33]
```diff
@@ -6,14 +6,17 @@
 
 .name-servers__dns {
 	.name-servers__toggle {
-		display: inline-block;
-		float: right;
-
-		.components-toggle-control .components-base-control__field .components-form-toggle {
+		.components-base-control__field {
+			justify-content: space-between;
 			margin-right: 0;
```
Comment (tyxla) [inline client/my-sites/domains/domain-management/name-servers/style.scss]: The right margin here won't do the job it did before. Note that with the latest change, `margin-right: 0;` is now declared for the `.components-base-control__field` selector, while previously it was for the `.components-toggle-control .components-base-control__field .components-form-toggle` selector.

**Review by tyxla — APPROVED** [review]
This LGTM and tests well 👍 

Thanks a bunch for the hard work here @spraveenitpro 🙌

**Comment by tyxla** [comment]
@spraveenitpro one last thing: the PR needs a rebase before we can land it. Thanks again 🙌
